testsuite/standalone/src/test/java/org/jboss/as/test/integration/management/cli/JBossCliXmlConfigTestCase.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.test.integration.management.cli;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamWriter;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import static org.junit.Assert.fail;
+import org.junit.Test;
+import org.wildfly.core.launcher.CliCommandBuilder;
+import org.wildfly.core.launcher.Launcher;
+
+/**
+ *
+ * @author Jean-François Denise <jdenise@redhat.com>
+ */
+public class JBossCliXmlConfigTestCase {
+
+    @Test
+    public void testSSLElement() {
+        testConfig("2.0");
+        testConfig("3.0");
+    }
+
+    private static void testConfig(String version) {
+        File f = createConfigFile(version);
+        execute(f);
+    }
+
+    private static void execute(File f) {
+        if (!f.exists()) {
+            fail("Config file " + f.getPath() + " doesn't exist");
+        }
+
+        final String jbossDist = TestSuiteEnvironment.getSystemProperty("jboss.dist");
+        if (jbossDist == null) {
+            fail("jboss.dist system property is not set");
+        }
+
+        final CliCommandBuilder commandBuilder = CliCommandBuilder.of(jbossDist);
+        commandBuilder.addJavaOptions("-Djboss.cli.config=" + f.toPath());
+
+        commandBuilder.addCliArgument("--command=help");
+        Process cliProc = null;
+        try {
+            cliProc = Launcher.of(commandBuilder)
+                    .setRedirectErrorStream(true)
+                    .launch();
+            cliProc.waitFor(2000, TimeUnit.MILLISECONDS);
```

Comment:
We have a util that we run all testuite timeouts through to make it easier to make them longer in odd environments. So just as a code standard thing this should be:

cliProc.waitFor(TimeoutUtil.adjust(2000), TimeUnit.MILLISECONDS);

org.jboss.as.test.shared.TimeoutUtil

Also, is 2000ms the appropriate value here? I think the test failure that was reported on the CI run is due to that amount being insufficient.

If the goal of the test isn't to check for something happening quickly, then use a real long timeout (like 30 secs). If the test fails but takes a long time to do so, that's ok; that should be a rare situation. OTOH intermittent CI failures due to short timeouts being exceeded are a real pain.